src/v/config/configuration.cc

```diff
@@ -1468,6 +1468,12 @@ configuration::configuration()
       "cloud_storage_cluster_metadata_upload_interval_ms",
       "Time interval to wait between cluster metadata uploads.",
       {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      1h)
+  , cloud_storage_cluster_metadata_upload_timeout_ms(
+      *this,
+      "cloud_storage_cluster_metadata_upload_timeout_ms",
+      "Timeout for cluster metadata uploads.",
```

Comment:
does this mean that uploads are failed after a timeout, or that metadata uploads shouldn't be more than timeout-seconds stale?

Reply:
This is a timeout to be used for individual metadata operational timeouts (e.g. sending an RPC with this timeout, uploading with this timeout, etc). It's not elegant at all, in that it's not a high-level timeout for the overarching process of uploading, but it's good enough for the interfaces we have.

Happy to fine tune this, I can imagine maybe passing around a deadline, or gradually reducing the timeout from callsite to callsite. For now it's quite simplistic.